cli/medperf/comms/entity_resources/utils.py

```diff
@@ -97,6 +102,9 @@ def download_resource(
 
     calculated_hash = verify_or_get_hash(tmp_output_path, expected_hash)
 
+    if calculated_hash is None:
+        raise InvalidEntityError(f"Hash mismatch: {resource}")
```

Comment:
`verify_or_get_hash` right now has quite a complex behavior depending on if `expected_hash` was passed:
- if passed, then function returns either `None` or `expected_hash` 
- if not passed, then returns `calculated_hash`

Thus, you have to split hash comparison logic onto two parts, here and inside `verify_or_get_hash`. What do you think if we simplify it, removing `verify_or_get_hash` totally:

```suggestion
    calculated_hash = get_file_hash(tmp_output_path)

    if expected_hash and expected_hash != calculated_hash:
        raise InvalidEntityError(f"Hash mismatch: {resource}, found: {calculated_hash}, expected: {expected_hash}")
```

Reply:
btw in which cases there may be no expected hash? Looks like it should always exist, isn't it?